modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase

```diff
@@ -749,6 +749,125 @@ definition {
+		property custom.properties = "feature.flag.LPS-165849=true";
+		property portal.acceptance = "true";
+
+		task ("Given two custom objects, A and B, each with a text field and publish them") {
+			for (var letters : list "A,B") {
+				ObjectAdmin.addObjectViaAPI(
+					labelName = "Custom Object ${letters} 182607",
+					objectName = "CustomObject${letters}182607",
+					pluralLabelName = "Custom Objects ${letters} 182607");
+
+				ObjectAdmin.addObjectFieldViaAPI(
+					fieldBusinessType = "Text",
+					fieldLabelName = "Field Text ${letters}",
+					fieldName = "customField",
+					fieldType = "String",
+					isRequired = "false",
+					objectName = "CustomObject${letters}182607");
+
+				ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject${letters}182607");
+			}
+		}
+
+		task ("And given a 1:M relationship between the Object A and Object B") {
+			ObjectAdmin.addObjectRelationshipViaAPI(
+				objectName_1 = "CustomObjectA182607",
+				objectName_2 = "CustomObjectB182607",
+				relationshipLabel = "Relationship",
+				relationshipName = "relationship",
+				relationshipType = "oneToMany");
+		}
+
+		task ("And given a User Custom with admin roles is added") {
+			JSONUser.addUser(
+				userEmailAddress = "userea@liferay.com",
+				userFirstName = "userfn",
+				userLastName = "userln",
+				userScreenName = "usersn");
+
+			JSONRole.assignRoleToUser(
+				roleTitle = "Administrator",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("And given a user notification template is created using terms from both object definitions") {
+			UserNotificationAPI.postNotificationTemplate(
+				en_US = "[%RELATIONSHIP_CUSTOMOBJECTA182607_CUSTOMFIELD%] | [%CUSTOMOBJECTB182607_CUSTOMFIELD%]",
+				externalReferenceCode = "ercTest",
+				name = "User Notification Template",
+				recipientType = "user",
+				type = "userNotification",
+				userScreenName = "usersn");
+		}
+
+		task ("And given the title field of Object A is changed") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectSystemObject(label = "Custom Object A 182607");
+
+			CreateObject.selectTitleField(fieldLabel = "Field Text A");
+
+			CreateObject.saveObject();
+		}
+
+		task ("And given a notification action is created on the Object 2 using the on after add trigger") {
+			Navigator.gotoBack();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object B 182607");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.addObjectActionViaUI(
+				actionLabel = "Notification action when entry is added",
+				active = "true",
+				notificationName = "User Notification Template",
+				objectLabel = "Custom Object B 182607",
+				thenAction = "Notification",
+				whenAction = "On After Add");
+		}
+
+		task ("And given the User is changed for the new User Custom") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "userea@liferay.com",
+				userLoginFullName = "userfn userln");
+		}
+
+		task ("When an object entry for both Objects") {
```

Comment:
Since you're using the value of the field of the Object A coming from the relationship, and not adding another entry on it, use: "When an entry in Object B is added"